### PR TITLE
P20-399: Create unit tests for Dashboard docs i18n sync-out

### PR DESCRIPTION
**Pull request by artem-vavilov**
1. Refactor the Dashboard `docs` resource `sync-in` business logic
2. Updated the Dashboard `docs` resource `sync-in` unit tests
3. Modularized the Dashboard `docs` resource `sync-out`
4. Created unit tests for the Dashboard `docs` resource `sync-out`
5. Created executable script `bundle exec bin/i18n/resources/dashboard/docs/sync_in.rb`
6. Created executable script `bundle exec bin/i18n/resources/dashboard/docs/sync_out.rb`

## Links
- jira ticket: [P20-399](https://codedotorg.atlassian.net/browse/P20-399)

### CODE REVIEW

File: bin/test/i18n/resources/dashboard/test_docs.rb [hunk 1]
```diff
@@ -1,112 +1,18 @@
...
-
-      # Should not be translated
-      generate_db_data_for('unexpected')
-      File.expects(:write).with(CDO.dir('i18n/locales/source/docs/unexpected.json'), expected_docs_json_for('unexpected')).never
-
+describe I18n::Resources::Dashboard::Docs do
+  describe '.sync_in' do
+    it 'sync-in Courses resource' do
+      I18n::Resources::Dashboard::Docs::SyncIn.expects(:perform).once
       I18n::Resources::Dashboard::Docs.sync_in
```
Comment (mgc1194) [inline bin/test/i18n/resources/dashboard/test_docs.rb:8]:
```suggestion
    it 'syncs in Docs' do

```

File: bin/test/i18n/resources/dashboard/test_docs.rb [hunk 2]
```diff
@@ -1,112 +1,18 @@
...
-        RedactRestoreUtils.expects(:redact).with(expected_source_file_path, expected_source_file_path, %w[visualCodeBlock link resourceLink]).in_sequence(exec_seq)
-      end
-
-      # Should not be translated
-      generate_db_data_for('unexpected')
-      File.expects(:write).with(CDO.dir('i18n/locales/source/docs/unexpected.json'), expected_docs_json_for('unexpected')).never
-
+describe I18n::Resources::Dashboard::Docs do
+  describe '.sync_in' do
+    it 'sync-in Courses resource' do
```
Comment (mgc1194) [inline bin/test/i18n/resources/dashboard/test_docs.rb]:
```suggestion
    it 'sync-in programming Docs' do
```

File: bin/test/i18n/resources/dashboard/test_docs.rb (hunk 1 above)
Comment (mgc1194) [inline bin/test/i18n/resources/dashboard/test_docs.rb:8]: is this line still necessary?

Comment (artem-vavilov) [reply bin/test/i18n/resources/dashboard/test_docs.rb:8]: What do you mean?

Comment (mgc1194) [reply bin/test/i18n/resources/dashboard/test_docs.rb:8]: Never mind, my brain was tricking me.

**Review by mgc1194 — APPROVED** [review]
I successfully tested sync-in and sync-out for Docs.

**Pull request merged**